test/support/query.rb

```diff
@@ -0,0 +1,20 @@
+# Test helpers to deal with query strings.
+
+def parse_query(query)
+  if query.nil?
+    {}
+  else
+    query_options query
+  end
+end
+
+def query_options(query)
+  options = {}
+
+  query.split('&').each do |pair|
+    k, v = pair.split('=')
+    options[k.to_sym] = v
```

Comment:
I was more concerned about query param parsing being fickle.  CGI will be reliable no matter what is added to the library in the future.